gensim/test/test_fasttext.py

```diff
@@ -1300,6 +1306,253 @@ def _run(self, fin):
+
+
+def _read_binary_file(fname):
+    with open(fname, "rb") as f:
+        data = f.read()
+    return data
+
+
+class SaveFacebookFormatFileGensimTest(unittest.TestCase):
+    """
+    This class containts tests that check the following scenario:
+
+    + create binary fastText file model1.bin using gensim
+    + load file model1.bin to variable `model`
+    + save `model` to model2.bin
+    + check if files model1.bin and model2.bin are byte identical
+    """
+
+    def _check_roundtrip_file_file(self, model_params):
+        with temporary_file("roundtrip_file_to_file1.bin") as fpath1, \
+            temporary_file("roundtrip_file_to_file2.bin") as fpath2:
+            _create_and_save_test_model(fpath1, model_params)
+            model = gensim.models.fasttext.load_facebook_model(fpath1)
+            gensim.models.fasttext.save_facebook_model(model, fpath2)
+            self.assertEqual(_read_binary_file(fpath1), _read_binary_file(fpath2))
+
+    def test_roundtrip_file_file_skipgram(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 1,
+            "negative": 0,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_file_file(model_params)
+
+    def test_roundtrip_file_file_cbow(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 0,
+            "negative": 0,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_file_file(model_params)
+
+
+def _save_test_model(out_base_fname, model_params, fasttext_cmd):
+    inp_fname = datapath('lee_background.cor')
+
+    model_type = "cbow" if model_params["sg"] == 0 else "skipgram"
+    size = str(model_params["size"])
+    seed = str(model_params["seed"])
+
+    cmd = fasttext_cmd + " " + model_type + " -input " + inp_fname + \
+            " -output " + out_base_fname + " -dim " + size + " -seed " + seed
+    subprocess.run(cmd, shell=True)
+
+
+class SaveFacebookFormatFileFastTextTest(unittest.TestCase):
+    """
+    This class containts tests that check the following scenario:
+
+    + create binary fastText file model1.bin using facebook_binary (FT)
+    + load file model1.bin to variable `model`
+    + save `model` to model2.bin using gensim
+    + check if files model1.bin and model2.bin are byte-identical
+
+    Requires env. variable FT_HOME to point to location of Facebook fastText binary
+    """
+
+    def _check_roundtrip_file_file(self, model_params):
+        ft_home = os.environ.get("FT_HOME", None)
+        fasttext_cmd = os.path.join(ft_home, "fasttext")
+
+        # fasttext tool creates both *vec and *bin files, so we have to remove both, even thought *vec is unused
+
+        with temporary_file("roundtrip_file_to_file1.bin") as fpath1bin, \
+            temporary_file("roundtrip_file_to_file2.bin") as fpath2bin, \
```

Comment:
I think if you indent by one more level, flake8 won't complain anymore.

But, even better, if you use shorter names, you won't need to span multiple lines with the context manager.

```python
with temp_file("model1") as m1, temp_file("model2"), temp_file("vector1") as v1:
```

After all, who are those long filenames for? Those files get nuked when the test exits, regardless of success or failure, so putting too much effort into naming them is not productive.

Reply:
Corrected.